spec/e2e/id_property_spec.rb

```diff
@@ -340,38 +366,153 @@ def foobar
     end
   end
 
+  describe 'id_property :neo_id' do
+    before do
+      before_session do
+        stub_const('NeoIdTest', UniqueClass.create do
+          include Neo4j::ActiveNode
+          id_property :neo_id
+        end)
+      end
+    end
+
+    it 'has an index' do
+      expect(NeoIdTest.mapped_label.indexes[:property_keys]).to be_empty
+    end
+
+    describe 'property id' do
+      it 'is not defined when before save ' do
+        node = NeoIdTest.new
+        expect(node.id).to be_nil
+      end
+
+      it 'is is set when saving ' do
+        node = NeoIdTest.new
+        node.save
+        expect(node.id).to be_present
+      end
+
+      it 'is same as neo_id' do
+        node = NeoIdTest.new
+        expect(node.id).to be_nil
+        node.save
+        expect(node.id).to eq(node.neo_id)
+      end
+    end
+
+    describe 'find_by_id' do
+      it 'finds it if it exists' do
+        NeoIdTest.create
+        node = NeoIdTest.create
+        NeoIdTest.create
+
+        found = NeoIdTest.find_by_id(node.id)
+        expect(found).to eq(node)
+      end
```

Comment:
I just copied 1 of the existing tests for uuid. I guess to ensure it does not find just a random node.